crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -430,81 +297,72 @@ pub(crate) fn do_finalize_slashed_operators<T: Config>(
 
             // transfer all the staked funds to the treasury account
             // any gains will be minted to treasury account
-            Nominators::<T>::drain_prefix(operator_id).try_for_each(|(nominator_id, _)| {
+            Deposits::<T>::drain_prefix(operator_id).try_for_each(|(nominator_id, deposit)| {
                 let locked_amount = T::Currency::balance_on_hold(&staked_hold_id, &nominator_id);
+
+                // if there was a pending deposit, deduct that amount from the slashable amount.
+                let (amount_to_slash, amount_to_release) =
+                    if let Some(pending_deposit) = deposit.pending {
```

Comment:
IIUC, the meaning of pending deposit is different from the previous version, in staking 1.1, we only move the `pending` deposit to `knowns` when there is a new deposit, which will become the new `pending` deposit, so `pending` may always be `Some` and the `pending` fund may already participated staking.

Reply:
Yes that is true and hence while slashing, we first convert any deposits that are from previous epoch which are already participating in staking. We do not slash pending deposits that happened in the epoch when operator was slashed. 

This is checked and fixed in the later commits